internal/pkg/cli/pipeline_init.go

```diff
@@ -374,24 +464,37 @@ func (o *initPipelineOpts) selectURL() error {
 // bbhttps	https://huanjani@bitbucket.org/huanjani/aws-copilot-sample-service.git (fetch)
 // bbssh	ssh://git@bitbucket.org:teamsinspace/documentation-tests.git (fetch)
 
-// parseGitRemoteResults returns just the trimmed middle column (url) of the `git remote -v` results,
-// and skips urls from unsupported sources.
-func (o *initPipelineOpts) parseGitRemoteResult(s string) ([]string, error) {
-	var urls []string
-	urlSet := make(map[string]bool)
+// parseGitRemoteResults returns just the first (shortname) and second (url) columns of the `git remote -v` results as a map (url: name), and skips urls from unsupported sources.
+func (o *initPipelineOpts) parseGitRemoteResult(s string) map[string]string {
+	repos := make(map[string]string)
 	items := strings.Split(s, "\n")
 	for _, item := range items {
 		if !strings.Contains(item, githubURL) && !strings.Contains(item, ccIdentifier) && !strings.Contains(item, bbURL) {
 			continue
 		}
 		cols := strings.Split(item, "\t")
-		url := strings.TrimSpace(strings.TrimSuffix(strings.Split(cols[1], " ")[0], ".git"))
-		urlSet[url] = true
+		URL := strings.TrimSpace(strings.Split(cols[1], " ")[0])
+		repos[URL] = cols[0]
+	}
+	return repos
+}
+
+func (o *initPipelineOpts) parseGitBranchResults(s string) ([]string, error) {
+	var branches []string
+	branchSet := make(map[string]bool)
+	items := strings.Split(s, "\n")
+	for _, item := range items {
+		elements := strings.Split(item, "/")
+		if len(elements) < 3 {
+			return nil, fmt.Errorf("unparsable format")
```

Comment:
```suggestion
			return nil, fmt.Errorf("cannot parse branch name from %s, item)
```

Reply:
I was trying to avoid the word "cannot" but I agree that this is better.